pkg/csi/csi_ops.go

```diff
@@ -83,11 +84,14 @@ func (o *validateOperations) ValidateVolumeSnapshotClass(ctx context.Context, vo
 type ApplicationCreator interface {
 	CreatePVC(ctx context.Context, args *types.CreatePVCArgs) (*v1.PersistentVolumeClaim, error)
 	CreatePod(ctx context.Context, args *types.CreatePodArgs) (*v1.Pod, error)
+	WaitForPVCReadyOrCheckEventIssues(ctx context.Context, namespace string, pvcName string) error
+	WaitForPodReadyOrCheckEventIssues(ctx context.Context, namespace string, podName string) error
 	WaitForPodReady(ctx context.Context, namespace string, podName string) error
 }
```

Comment:
@bathina2 or maybe better option than adding these 2 to the interface would be replacing `WaitForPodReady` content with calls to `waitForPVCReadyOrCheckEventIssues/Pod`...or add only `WaitForPVCReady` and replace the content of Pod with this new addition